lib/omnibus/config.rb

```diff
@@ -333,6 +333,13 @@ def reset!
     # @return [Integer]
     default(:publish_retries, 2)
 
+    # Directory pattern for publisher. Interpolation of matadata keys are supported.
+    #
+    # @example 'custom_prefix/%{platform}/%{platform_version}/%{arch}/%{basename}'
+    #
+    # @return [Integer]
+    default(:publish_dir_pattern, nil)
```

Comment:
Ah, I see. I've missed that. Would it make sense to have two different settings then? `artifactory_publish_pattern` and `s3_publish_pattern`? (No strong opinion, just floating ideas.)